src/sparseml/core/logger/utils/frequency_managers.py

```diff
@@ -0,0 +1,209 @@
+    ):
+        """
+        :param current_log_step: The current log step
+        :param last_log_step: The last step we logged at
+        :return: True if the frequency manager is ready to log,
+            False otherwise
+        """
+        return self._log_frequency is not None and (
+            current_log_step is None
+            or last_log_step is None
+            or current_log_step >= last_log_step + self._log_frequency
+        )
+
+    def validate_log_frequency(self, log_frequency: Union[float, int, None]):
+        """
+        Validates the log frequency is a float or int or None, raising
+        a ValueError if invalid
+
+        :param log_frequency: _description_
+        :raises ValueError: _description_
+        """
+        if not isinstance(log_frequency, (float, int, type(None))):
+            raise ValueError(
+                f"frequency {log_frequency} must be a float or int or None"
+                f" But found {type(log_frequency)} instead"
+            )
+
+    @property
+    def log_frequency(self) -> Union[float, int, None]:
+        return self._log_frequency
+
+    @log_frequency.setter
+    def log_frequency(self, value: Union[float, int, None]):
+        self.validate_log_frequency(log_frequency=value)
+        self._log_frequency = value
+
+
+class IntegerFrequencyManager_(EpochFrequencyManager):
+    """
+    Frequency manager that handles logging based on integer steps only.
+    """
+
+    def validate_log_frequency(self, log_frequency: Optional[int]):
+        """
+        Validates the log frequency is an int or None, raising a ValueError if invalid
+
+        :param log_frequency: the log frequency to validate
+        :raises ValueError: if the log frequency is neither an int or None
+        """
+        if not isinstance(log_frequency, (int, type(None))):
+            raise ValueError(f"frequency {log_frequency} must be an int or None")
+        super().validate_log_frequency(log_frequency)
+
+
+class BatchFrequencyManager(IntegerFrequencyManager_):
+    """
+    Frequency manager that handles logging based on batches.
+    Accepts an int for the frequency, where the frequency is the number of batches
+    between logs, or None for to log on every batch. If not None, the frequency
+    must be >= 16
+    """
+
+    def validate_log_frequency(self, log_frequency: Optional[int]):
+        """
+        Validates the log frequency is an int or None, raising a ValueError if invalid
+        Also validates that the frequency is >= 100 (to avoid over-logging) if not None
+
+        :param log_frequency: the log frequency to validate
+        :raises ValueError: if the log frequency is neither an int or None
+        :raises ValueError: if the log frequency is < 10
+        """
+        super().validate_log_frequency(log_frequency)
+        if log_frequency is not None and log_frequency < 10:
+            raise ValueError(f"frequency {log_frequency} must be >= 10")
```

Comment:
docstrings are inconsistent with the code, class definition says >=16, fn definition says <100, actual code says >=10. Also could you clarify why we don't allow < 10?

Reply:
It is done to avoid over-logging; however I think we should remove this constraint all together and chalk it up to user error if a lower frequency is specified

Reply:
Agreed! if they really want logging every batch thats their problem :)